pkg/v1/resource/chartconfig/resource.go

```diff
@@ -112,6 +112,29 @@ func containsChartConfig(list []*v1alpha1.ChartConfig, item *v1alpha1.ChartConfi
 	return false
 }
 
+func getChartConfigByName(list []*v1alpha1.ChartConfig, name string) (*v1alpha1.ChartConfig, error) {
+	for _, l := range list {
+		if l.Name == name {
+			return l, nil
+		}
+	}
+
+	return nil, microerror.Mask(notFoundError)
+}
+
+func isChartConfigModified(a, b *v1alpha1.ChartConfig) bool {
+	// If the Spec section has changed we need to update.
+	if !reflect.DeepEqual(a.Spec, b.Spec) {
+		return true
+	}
+	// If the Labels have changed we also need to update.
+	if !reflect.DeepEqual(a.Labels, b.Labels) {
+		return true
+	}
+
+	return false
+}
```

Comment:
The chartconfig CR's are versioned by the cluster-operator version bundle. Since the Spec section holds the chart config if there is any change we update the CR in the guest cluster.

We also update if the labels change so we can propagate metadata changes.